pkg/operator/ceph/file/mds/spec.go

```diff
@@ -155,7 +155,7 @@ func (c *Cluster) makeMdsDaemonContainer(mdsConfig *mdsConfig) v1.Container {
 }
 
 func (c *Cluster) podLabels(mdsConfig *mdsConfig, includeNewLabels bool) map[string]string {
-	labels := controller.CephDaemonAppLabels(AppName, c.fs.Namespace, "mds", mdsConfig.DaemonID, includeNewLabels)
+	labels := controller.CephDaemonAppLabels(AppName, c.fs.Namespace, config.MdsType, mdsConfig.DaemonID, mdsConfig.ResourceName, "cephfilesystems.ceph.rook.io", includeNewLabels)
```

Comment:
The CR name should generally be on the cluster properties something like this:
```suggestion
	labels := controller.CephDaemonAppLabels(AppName, c.fs.Namespace, c.fs.Name, config.MdsType, mdsConfig.DaemonID, mdsConfig.ResourceName, "cephfilesystems.ceph.rook.io", includeNewLabels)
```